### DISCUSSION TITLE
I can't build Chocolatey wtih a Cake exception

### DISCUSSION THREAD

**Discussion by buzhi-kefou** [cake-build/cake#4161] [Q&A] [answered]
how can i solve this error?
<img width="960" alt="4DC40488-C4E7-471c-BE8D-63446FF9835A" src="https://github.com/cake-build/cake/assets/61810948/cc190627-25c2-4b37-b750-4011c6d099fc">

**Comment by gep13** [comment]
Can you confirm if you have cloned the Chocolatey CLI repository using git, or have you downloaded the zip file of the source code?

**Reply by buzhi-kefou** [reply]
i downloaded the zip file of the source code

**Reply by buzhi-kefou** [reply]
this is my git config of Chocolatey repository， i can't understand the error with ’No commits found on the current branch‘
<img width="827" alt="62738B74-A092-450a-9AAA-717B8662F24E" src="https://github.com/cake-build/cake/assets/61810948/c971fe61-a7cd-458a-a986-4e3ed15a72f1">

**Reply by gep13** [reply, marked as answer]
The build for Chocolatey CLI uses a tool called GitVersion in order to assert a semantic version for the current state of the repository.  In order for that system to work, it needs the entire history of the git repository to be available, i.e. you need to clone the entire repository, not simply download the zip file of the current source code.

**Reply by buzhi-kefou** [reply]
Thank you, I completed this process successfully.
